### ISSUE TITLE
Transitive dependencies with classifier are not downloaded

### ISSUE THREAD

**Issue by joan38** [sbt/sbt#5136]
## steps
sbt version: 1.3.2

I think I've been able to reproduce this outside of Netflix:
```
libraryDependencies += "io.netty" % "netty" % "3.10.6.Final"
```
Then `show Runtime/fullClasspath` won't show `netty-tcnative` whereas we depend on it in https://repo1.maven.org/maven2/io/netty/netty/3.10.6.Final/netty-3.10.6.Final.pom

## problem

I have a dependency that depends transitively as the following ivy file shows:
```
<dependency org="com.netflix.bdp" name="presto-s3fs" rev="1.3.2" conf="compile->default">
  <artifact xmlns:m="http://ant.apache.org/ivy/maven" name="presto-s3fs" type="jar" ext="jar" m:classifier="core"/>
</dependency>
```
When `useCoursier := true` that transitive dependency is not in the classpath.

## workaround

`useCoursier := false`
or explicitly add:
`("com.netflix.bdp" % "presto-s3fs" % "1.3.2").classifier("core")`

**Comment by eed3si9n** [comment]
@joan38 Thanks for the report. Could you also report this to coursier/coursier as well please?

**Comment by dwijnand** [comment]
Fixed in 1.3.4 with the upgrade to (lm-) Coursier 2.0.0-RC5-2.
